README.md

```diff
@@ -63,1498 +104,468 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
-
-## 7.1 Schema
-
-```ipldsch
-# Represents a request to invoke enclosed set of tasks concurrently
-type Effects {
-  # Primary set of tasks to be invoked
-  fork      [&Invocation]
-  
-  # Additional task to be invoked with added semantics
-  # of representing a workflow execution continuation.
-  join      optional &Invocation
-}
+})
 ```
 
-## 7.2 Fields
-
-### 7.2.1 Forked Task Invocations
+### Agents
+[Agents]: #agents
 
-The OPTIONAL `fork` field, if present MUST be a list of an alphabetically ordered [Task] links. List MUST NOT not contain duplicate entries.
+#### Issuer
+[Issuer]: #issuer
 
-### 7.2.2 Joined Task Invocation
+The `iss` field MUST include the Issuer of the Invocation. This DID MUST match against the encoding signature.
 
-The OPTIONAL `join` field, if present MUST be set to a [Task] link.
+#### Subject
+[Subject]: #subject
 
-## 7.3 DAG-JSON Examples
+The REQUIRED `sub` field both parameterizes over a specific agent, and acts as a namespace for how to interpret the [Command]. This is especially critical for two parts of the life cycle:
 
-### 7.3.1 Effect spawning concurrent threads
-
-```json
-{
-  "fork": [
-    {
-      "/": "bafyreigmmdzix2vxboojvv6j6h7sgvxnrecdxtglwtqpxw7hybebzlsax4"
-    },
-    {
-      "/": "bafyreif6gfpzgxnii4ys6a4bjenefg737fb5bgam3onrbmhnoa4llk244q"
-    }
-  ]
-}
-```
+1. Specifying a particular `sub` (and thus `aud`) when [enqueuing new Tasks][enqueue] in a Receipt
+2. Indexing Receipts for reverse lookup and memoization
 
-### 7.3.2 Effect continuing thread execution
+#### Audience
+[Audience]: #audience
 
-```json
-{
-  "join": {
-    "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-  }
-}
-```
+The OPTIONAL `aud` field specified the intended recipient of Invocation, otherwise the Audience MUST be assumed to the [Subject]. This is useful for message routing, command brokers, proxy execution, gateways, replicated state machines, and so on.
 
-### 7.3.1 Effect with fork & join
+### Task
+[Task]: #task
 
-```json
-{
-  "join": {
-    "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-  },
-  "fork": [
-    {
-      "/": "bafyreigmmdzix2vxboojvv6j6h7sgvxnrecdxtglwtqpxw7hybebzlsax4"
-    },
-    {
-      "/": "bafyreif6gfpzgxnii4ys6a4bjenefg737fb5bgam3onrbmhnoa4llk244q"
-    }
-  ]
-}
-```
-
-# 8 Receipt
-
-A `Receipt` is an attestation of the [Result] and requested [Effect]s by a [Task] [Invocation]. A Receipt MUST be signed by the [Executor] or it's delegate. If signed by the delegate, the proof of delegation from the [Executor] to the delegate (the `iss` of the receipt) MUST be provided in `prf`.
+A Task is the subset of Invocation fields that uniquely determine the work to be performed. The nonce is important for distinguishing between non-idempotent executions of a Task by making the group together unique.
+ 
+A Task MUST be uniquely defined by a Task ID that is the CID of the following fields:
```

Comment:
of the... CONCATENATION of these fields? of a flat structure of all their values? I think being a little more explicit in how these fields get canonicalized before get magically CID-ified would go a long way here.

Reply:
It's the CID of those fields as a struct. Will fix.